src/main/java/org/folio/helper/BindHelper.java

```diff
@@ -66,6 +71,39 @@ private Map<String, Map<String, BindPiecesCollection>> groupBindPieceByPoLineId(
     return Map.of(poLineId, bindPieceMap);
   }
 
+  public Future<Void> removeBinding(String pieceId, RequestContext requestContext) {
+    logger.debug("removeBinding:: Removing binding for piece: {}", pieceId);
+    return pieceStorageService.getPieceById(pieceId, requestContext)
+      .compose(piece -> {
+        var bindItemId = piece.getBindItemId();
+        piece.withBindItemId(null).withIsBound(false);
+        return removeForbiddenEntities(piece, requestContext)
+          .compose(v -> getValidPieces(requestContext))
+          .compose(piecesGroupedByPoLine -> storeUpdatedPieceRecords(piecesGroupedByPoLine, requestContext))
+          .compose(piecesGroupedByPoLine -> clearTitleBindItemsIfNeeded(piece.getTitleId(), bindItemId, requestContext));
+      });
+  }
+
+  private Future<Void> removeForbiddenEntities(Piece piece, RequestContext requestContext) {
+    // Populate piecesByLineId used by removeForbiddenEntities and parent helper methods
+    piecesByLineId = Map.of(piece.getPoLineId(), Collections.singletonMap(piece.getId(), null));
+    return removeForbiddenEntities(requestContext);
+  }
+
+  private Future<Void> clearTitleBindItemsIfNeeded(String titleId, String bindItemId, RequestContext requestContext) {
+    String query = String.format("titleId==%s and bindItemId==%s and isBound==true", titleId, bindItemId);
+    return pieceStorageService.getPieces(0, 0, query, requestContext)
+      .compose(pieceCollection -> {
+        var totalRecords = pieceCollection.getTotalRecords();
+        logger.info("clearTitleBindItemsIfNeeded:: Found {} pieces associated with bind item {}", bindItemId, totalRecords);
+        if (totalRecords != 0) {
+          return Future.succeededFuture();
+        }
+        return titlesService.getTitleById(titleId, requestContext)
+          .compose(title -> titlesService.saveTitle(title.withBindItemIds(List.of()), requestContext));
```

Comment:
it seems we are clearing all bindItemIds, we should remove only required bindItemId.

```
return titlesService.getTitleById(titleId, requestContext)
          .compose(title -> {
              List<String> bindItemIds = new ArrayList<>(title.getBindItemIds());
              bindItemIds.remove(bindItemId);
              return titlesService.saveTitle(title.withBindItemIds(bindItemIds), requestContext);
          });
```